uwblueprint/focus-on-nature
# feat: payments

## Notion ticket link

[Payments - Stripe API](https://uwblueprintexecs.notion.site/Payments-Stripe-API-945afc1e153d472da2c31ffc79286723)



## Implementation description
* created a stripe client to handle creating stripe checkout session 
* call createCheckoutSession function from stripe client in camperRouter's /register endpoint 
* log the checkout session URL 
* created a webhook in camperRouter to listen to stripe events 
* webhook calls various functions depending on the event passed e.g. `checkout.session.completed`



## Steps to test
1. run `stripe listen --api-key <SECRET_KEY> --forward-to localhost:5000/camper/webhook` in a terminal to listen to stripe events
2. make a post request on http://localhost:5000/campers/register 
3. click checkout session link that's logged in terminal upon a successful post request 
4. fill out payment information and submit the payment 
5. check in stripe terminal if the correct events are properly logged 


## What should reviewers focus on?
* are all the components in the correct files 
* is there any code that could be simplified


## Checklist
